source/common/compressor/zlib_compressor_impl.cc

```diff
@@ -42,10 +42,10 @@ void ZlibCompressorImpl::compress(Buffer::Instance& buffer, bool trailer) {
     zstream_ptr_->avail_in = input_slice.len_;
     zstream_ptr_->next_in = static_cast<Bytef*>(input_slice.mem_);
     process(buffer, Z_NO_FLUSH);
+    buffer.drain(input_slice.len_);
```

Comment:
I see that you found a way to use no_flush. Did you check if it works with multiple slices?

Reply:
I have a test in here a291b5c.